### PR TITLE
Makes the github PR workflow test against all OSes

### PR DESCRIPTION
**Pull request by m-rph**
Fixes: #153 

_Description of changes_: makes the github PR workflow test against all OSes. Fixing integ tests is out of scope of this PR.
Drops 3.7 from all OSes except for AL1. AL1 will be removed completely in subsequent PR.

_Target (OCI, Managed Runtime, both)_: N/A

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by krk — APPROVED** [review]
thanks, lgtm

File: .github/workflows/test-on-push-and-pr.yml [hunk 1]
```diff
@@ -12,11 +12,45 @@ jobs:
...
+    - name: Run debian integration tests
+      run: DISTRO=debian make test-integ
+
+  ubuntu:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Run ubuntu integration tests
+      run: DISTRO=ubuntu make test-integ
```
Comment (m-rph) [inline .github/workflows/test-on-push-and-pr.yml:56]: This was pulled directly from the nodejs ric.

File: Makefile [hunk 2]
```diff
@@ -21,7 +21,7 @@ test-smoke: setup-codebuild-agent
 
 .PHONY: test-integ
 test-integ: setup-codebuild-agent
-	CODEBUILD_IMAGE_TAG=codebuild-agent tests/integration/codebuild-local/test_all.sh tests/integration/codebuild/.
+	CODEBUILD_IMAGE_TAG=codebuild-agent DISTRO="$(DISTRO)" tests/integration/codebuild-local/test_all.sh tests/integration/codebuild/.
```
Comment (m-rph) [inline Makefile:24]: Pulled from nodejs ric, with some modification on the test path.

File: tests/integration/codebuild-local/test_all.sh [hunk 3]
```diff
@@ -51,10 +52,12 @@ main() {
         usage
         exit 1
     fi
-
+    
     BUILDSPEC_YML_DIR="$1"
+    echo $DISTRO $BUILDSPEC_YML_DIR
+    ls $BUILDSPEC_YML_DIR
```
Comment (m-rph) [inline tests/integration/codebuild-local/test_all.sh:58]: Nit: this isn't necessary, will remove later.

**Pull request merged**